packet_analysis/equation/generate_equation.py

```diff
@@ -0,0 +1,130 @@
+
+
+def get_segmented_average(conn: sqlite3.Connection, table: str, adjust_for_zero=False, segments=2):
+    """
+    Create lists of averages over a 24 hour time period, segmented if desired
+    :param conn: Database connection
+    :param table: Table to pull data from
+    :param adjust_for_zero: Adjust for zero in calculations (Prevents unrealistic averages in low quantity data sets)
+    :param segments: Number of segments (Data points) per hour
+    :return: List of averages, list of deviations
+    """
+    collection = {}
+    averaged_data = []
+    averaged_deviation = []
+    total_days = set()
+
+    # Loop through the hours in a day
+    for x in range(24):
+        # Loop through the segments
+        for y in range(segments):
+            segment = round(y / segments, 2)
+
+            # Set bounds of the segment
+            lower_bound = x + segment
+            upper_bound = x + round((y + 1) / segments, 2)
+            values = get_values(
+                conn, table, ["day"], ["hour >= {0}".format(lower_bound),
+                                       "hour < {0}".format(upper_bound)])
+
+            # Loop through the list of returned values, separate by day
+            days = set(values)
+            total_days |= days
+            if adjust_for_zero:
+                collection[x + segment] = []
+
+            # Loop through the days, create a collection keyed on segment time, filled with data from each day
+            for day in days:
+                count = values.count(day)
+                key = x + segment
+                if key in collection:
+                    collection[key].append(count)
+                else:
+                    collection[key] = [count]
+
+    # Loop through the values added to the collection
+    for x in collection:
+        segment_values = collection[x]
+        segment_values_length = len(segment_values)
+
+        # Pack with zeros for missing days, in order to get a true average
+        if segment_values_length < len(total_days):
+            for y in range(len(total_days) - segment_values_length):
+                segment_values.append(0)
+
+        # Calculate the mean and deviation
+        mean = statistics.mean(segment_values)
+        deviation = statistics.pstdev(segment_values)
+
+        # Append this information as tuples [(x, y) values]
+        averaged_data.append((x, mean))
+        averaged_deviation.append((x, deviation))
+
+    # Return lists of (x, y) data for average and deviation
+    return averaged_data, averaged_deviation
+
+
+# Seventh degree polynomial function
+def objective(x, a, b, c, d, e, f, g, h):
+    """
+    Objective function for the curve_fit to use
+    :param x: Input to function
+    :param a, b, c, d, e, f, g, h: Coefficients of sub expressions
+    :return: Value calculated from function
+    """
+    return (a * x) + (b * x ** 2) + (c * x ** 3) + (d * x ** 4) + (e * x ** 5) + (f * x ** 6) + (g * x ** 7) + h
```

Comment:
Wowza

Reply:
It's a long boi, but the longer (more degrees) the more accurate.